File: FriendMusic/FriendMusic/Models/Person.cs
using System.Text.Json.Serialization;

namespace FriendMusic.Models
{
    public class Person
    {
        public int Id { get; set; }
        
        public string FirstName { get; set; } = String.Empty;
        
        public string LastName { get; set; } = String.Empty;
        
        public DateTime Birthday { get; set; } = new DateTime();

        public Song? FavoriteSong { get; set; }

        public List<Playlist>? LikedPlaylists { get; set; }

        public List<Playlist>? OwnedPlaylists { get; set; }
    }
}

File: FriendMusicWeb/friend-music-web/src/models/person.ts
import Playlist from "./playlist";
import Song from "./song";

export interface Person {
    Id: number;
    FirstName: string;
    LastName: string;
    Birthday: string;
    FavoriteSong: Song;
    LikedPlaylists: Playlist[];
    OwnedPlaylists: Playlist[];
}

export interface PersonToAdd {
    FirstName: string;
    LastName: string;
    Birthday: string;
}

export interface PersonToUpdate {
    Id: number;
    FirstName: string;
    LastName: string;
    Birthday: string;
}



export default Person;